packages/webapp/src/containers/Help/index.jsx

```diff
@@ -1,30 +1,34 @@
-import React from 'react';
 import PureHelpRequestPage from '../../components/Help';
 import { useDispatch, useSelector } from 'react-redux';
 import { supportFileUpload } from './saga';
-import history from '../../history';
 import { isHelpLoadingSelector, startSendHelp } from '../Home/homeSlice';
 import { userFarmSelector } from '../userFarmSlice';
+import { useState } from 'react';
 
-export default function HelpRequest() {
+export default function HelpRequest({ closeDrawer }) {
   const dispatch = useDispatch();
 
   const handleSubmit = (file, data) => {
     dispatch(startSendHelp());
     dispatch(supportFileUpload({ file, form: data }));
   };
-  const handleBack = () => {
-    history.push('/');
+  const onCancel = () => {
+    closeDrawer?.();
+    setKey((prev) => prev + 1); // Remount form when closing drawer
```

Comment:
This was not my first choice, but `reset()` on the form populated the error object (and so all the validation error messages), which persisted upon opening the form again. I tried all the [properties on reset](https://react-hook-form.com/docs/useform/reset) before deciding to remount.

Reply:
`reset` actually works properly here because it does reset the form state in its entirety. The problem is on the rendering logic. If we don't do this key update, when the form is cleared and the values change, the component will rerender, but it won't remount. Every time the form rerenders, React Hook Form will re calculate the errors (so the problem isn't solved by clearing out the errors when canceling, either). The key update fixes the problem, but in a drastic way -- it just unmounts the whole thing, remounts and starts over.

I think the correct way to solve this is to fix the error rendering logic in the component by only showing the error messages if the fields are actually dirty, e.g.

```js
        {errors[SUPPORT_TYPE] && SUPPORT_TYPE in dirtyFields ? (
          <Error>{t('HELP.REQUIRED_LABEL')}</Error>
        ) : (
          ''
        )}
```

where `dirtyFields` is a property in `formState`.

You'll also need to set a defaultValue of `null` for the support type field so that it resets and clears out properly, since `undefined` won't work.

Reply:
> Since reset is using the same defaultValues, it seems there should be a way to get the full form state (errors inclusive) to the original pristine state, as if remounting.

Oh my goodness @SayakaOno solved the mystery -- the cancel button was submitting 🙇 🙇 🙇 🙏🙏